### PR TITLE
example: add microsoft/swin-large-patch4-window7-224

### PR DESCRIPTION
**Pull request by xieofxie**
uv run ~\ModelKit\examples\microsoft-swin-large-patch4-window7-224\example.py --onnx ~\.cache\winml\artifacts\microsoft_swin-large-patch4-window7-224\imgcls_ec485f4653d962b9_quantized.onnx
True label:  house finch, linnet, Carpodacus mexicanus (synset=n01532829, id=12)

Top 5 predictions:
  1. house finch, linnet, Carpodacus mexicanus (0.9127)
  2. brambling, Fringilla montifringilla (0.0122)
  3. goldfinch, Carduelis carduelis (0.0028)
  4. chickadee (0.0013)
  5. junco, snowbird (0.0013)

Verdict (top-1): PASS

Annotated image written to prediction.png

### CODE REVIEW

File: examples/microsoft_swin-large-patch4-window7-224/example.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        type=Path,
+        default=Path("prediction.png"),
+        help="Where to write the annotated image (default: prediction.png).",
+    )
+    return parser.parse_args()
+
+
+def load_image(image_arg: Path | None) -> tuple[Image.Image, str | None]:
+    """Load an image and (when streamed from the eval dataset) its WordNet synset.
+
```
Comment (DingmaomaoBJTU) [inline examples/microsoft_swin-large-patch4-window7-224/example.py:89]: The `load_image` function calls `load_dataset(DEFAULT_DATASET, split=DEFAULT_DATASET_SPLIT)` without `streaming=True` but the comment says `streaming=False so we can read the ClassLabel`. This downloads the entire dataset just to read one sample. Consider using `streaming=True` and extracting features separately, or at minimum add `trust_remote_code=False` for safety.

**Review by DingmaomaoBJTU — COMMENTED** [review]
Nice example! The README is well-structured with clear step-by-step instructions and the comparison table is very helpful. The example.py script is thorough with good error handling. One minor suggestion left inline about dataset loading efficiency. Also note: the directory rename from `microsoft-table-transformer-detection` to `microsoft_table-transformer-detection` is a good consistency fix but could be called out in the PR description.

**Review by DingmaomaoBJTU — COMMENTED** [review]
Nice example addition! The README walkthrough is clear and well-structured (config -> build -> eval -> perf -> PyTorch baseline -> comparison table). The `example.py` is thorough with good docstrings and handles both local images and dataset streaming.

The directory rename from hyphens to underscores for the existing example is a good consistency fix.

A couple of minor suggestions inline.

File: examples/microsoft_swin-large-patch4-window7-224/example.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    label_feature = dataset.features.get("label")
+    if label_value is None or label_feature is None or not hasattr(label_feature, "names"):
+        return image, None
+    return image, label_feature.names[int(label_value)]
+
+
+def imagenet_synset_to_id() -> dict[str, int]:
+    """Map WordNet synset ID -> ImageNet-1k class id (0-999).
+
+    Uses ``timm.data.ImageNetInfo`` so we don't have to ship the 1000-entry
```
Comment (DingmaomaoBJTU) [inline examples/microsoft_swin-large-patch4-window7-224/example.py:128]: **Suggestion:** `load_dataset(..., split=DEFAULT_DATASET_SPLIT)` without `streaming=True` downloads the entire dataset just to use `dataset[0]`. Consider `streaming=True` + `next(iter(dataset))` to avoid downloading ~3GB for a single sample, or add a comment explaining why streaming is intentionally avoided (e.g., needing `.features` metadata).

File: examples/microsoft_swin-large-patch4-window7-224/example.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    outputs = model(pixel_values=inputs["pixel_values"])
+
+    # logits: (1, num_classes). softmax → probabilities, then top-k.
+    logits = outputs.logits
+    probs = torch.softmax(logits, dim=-1)[0]
+    top_k = min(args.top_k, probs.numel())
+    top_scores, top_ids = torch.topk(probs, k=top_k)
+
+    # WinML's bare-ONNX path doesn't attach an HF config to the model, so
+    # pull id2label from the HF hub for human-readable label names.
```
Comment (DingmaomaoBJTU) [inline examples/microsoft_swin-large-patch4-window7-224/example.py:208]: Nit: `input_shapes` could be an empty list or have fewer than 4 elements for non-image models if someone copies this script as a template. The guard `if len(input_shapes) == 4` handles it, but a brief comment like `# Only applies to 4D image inputs (B,C,H,W)` would help copy-pasters.

**Review by DingmaomaoBJTU — COMMENTED** [review]
Well-written example with clear step-by-step docs and a comprehensive Python script. The README table showing the NPU vs CPU comparison (0.836 vs 0.837 accuracy, 64.9ms vs 662.3ms latency, 193MB vs 750MB) is a great selling point. The directory rename consistency fix for table-transformer is a nice bonus. LGTM.

File: examples/microsoft_swin-large-patch4-window7-224/example.py (hunk 1 above)
Comment (xieofxie) [reply examples/microsoft_swin-large-patch4-window7-224/example.py:89]: Good catch!

File: examples/microsoft_swin-large-patch4-window7-224/example.py [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+
+def imagenet_synset_to_id() -> dict[str, int]:
+    """Map WordNet synset ID -> ImageNet-1k class id (0-999).
+
+    Uses ``timm.data.ImageNetInfo`` so we don't have to ship the 1000-entry
+    list inline. The mapping is the canonical ImageNet-1k ordering that
+    the model was trained against.
+    """
+    from timm.data import ImageNetInfo
```
Comment (DingmaomaoBJTU) [inline examples/microsoft_swin-large-patch4-window7-224/example.py]: The `datasets` import is deferred (good for optional dep), but `timm.data.ImageNetInfo` in `imagenet_synset_to_id()` (line 153) is also a runtime-only dep not listed in the script's top-level imports. Consider adding a brief docstring note or a try/except with a clear error message if `timm` is not installed, similar to how `datasets` is handled implicitly.

File: examples/microsoft_swin-large-patch4-window7-224/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  -o build_config.json
+```
+
+```powershell
+winml build `
+  -c build_config.json `
+  -m microsoft/swin-large-patch4-window7-224 `
+  --device npu `
+  --ep openvino `
+  --use-cache
```
Comment (DingmaomaoBJTU) [inline examples/microsoft_swin-large-patch4-window7-224/README.md:34]: Nit: The README says `--ep openvino` but this is a Swin Transformer model. Is OpenVINO the only tested EP? Would be helpful to note if QNN was also tested or if there's a known limitation (e.g., unsupported ops) that restricts this to OpenVINO.

**Review by DingmaomaoBJTU — COMMENTED** [review]
LGTM overall. Clean example with good documentation. The README walkthrough is easy to follow and the comparison table is a nice addition. A couple minor nits.

File: examples/microsoft_swin-large-patch4-window7-224/README.md (hunk 5 above)
Comment (xieofxie) [reply examples/microsoft_swin-large-patch4-window7-224/README.md:34]: by design. just openvino ep

File: examples/microsoft_swin-large-patch4-window7-224/example.py (hunk 4 above)
Comment (xieofxie) [reply examples/microsoft_swin-large-patch4-window7-224/example.py]: fixed!

**Review by DingmaomaoBJTU — APPROVED** [review]
